rmgpy/rmg/model.py

```diff
@@ -687,6 +687,13 @@ def enlarge(self, newObject=None, reactEdge=False,
             logging.info('Generating thermo for new species...')
             self.applyThermoToSpecies(procnum)
 
+        # Do thermodynamic filtering
+        try:
+            if not numpy.isinf(self.toleranceThermoKeepSpeciesInEdge) and spcs != []:
```

Comment:
I think `spcs` is not properly defined here - replacing it with `self.newSpeciesList` should give the intended behavior.

Reply:
done